src/app/components/common/shapes/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { IShapesLibraryPlugin } from "../../../../compiler/runtime/libraries/shapes";
+import { Shape } from "../../../shapes/Shape";
+import { Rectangle } from "../../../shapes/Rectangle";
+import { Ellipse } from "../../../shapes/Ellipse";
+import { Triangle } from "../../../shapes/Triangle";
+import { Line } from "../../../shapes/Line";
+import { TextShape } from "../../../shapes/Text";
+
+export class ShapesComponent /*extends React.Component<Props, State>*/ implements IShapesLibraryPlugin {
+    private nameGenerationCounter: number;
+    private shapes: {[name: string] : Shape };
+
+    public constructor(){
+        this.nameGenerationCounter = 0;
+        this.shapes = {};
+    }
+
+    private generateName(shapeType: string): string {
+        this.nameGenerationCounter ++;
+        return shapeType + this.nameGenerationCounter.toString;
+    }
+
+    public addRectangle(width: number, height: number): string {
+        const name = this.generateName("Rectangle");
+        let obj = new Rectangle(name, width, height);
+        this.shapes[name] = obj;
+        return name;
+    }
+
+    public addEllipse(width: number, height: number): string {
+        const name = this.generateName("Ellipse");
+        let obj = new Ellipse(name, width, height);
```

Comment:
I'm not sure `obj` is the best variable name for these. Why not just call them the actual shape they are? i.e. `const ellipse = ...`